backend/src/main.rs

```diff
@@ -512,6 +512,19 @@ async fn user_connected(ws: WebSocket, games: Games, stats: Arc<Mutex<InMemorySt
             };
             info!(game.tracer(&logger, &room, Some(1)), "Joining room"; "player_id" => player_id.0);
             game.users.insert(ws_id, UserState { player_id, tx });
+
+            // if the same user joined before, remove its previous entry from the user list
+            if game
+                .game
+                .dump_state()
```

Comment:
Can we add an accessor function for this field directly rather than cloning the entire game structure?